lib/search.rb

```diff
@@ -265,6 +265,16 @@ def self.advanced_filters
     @advanced_filters
   end
 
+  advanced_filter(/^in:tagged$/) do |posts|
+    posts
+      .where('EXISTS (SELECT 1 FROM topic_tags WHERE topic_tags.topic_id = posts.topic_id)')
+  end
+
+  advanced_filter(/^in:untagged$/) do |posts|
+    posts
+      .where('NOT EXISTS (SELECT 1 FROM topic_tags WHERE topic_tags.topic_id = posts.topic_id)')
```

Comment:
sorry ... last one... leave this as a left join, it is certainly the fastest way of doing it and will not expand the result set like the opposite does... merging it right after that .

Reply:
:smile: no worries. Done, tested, & pushed. Let me know if you want me to collapse these down.